accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2746,45 +2746,75 @@ To match a pattern `p` against a value `v`:
         *This type test may get elided. See "Pointless type tests and legacy
         types" below.*
 
-    2.  Let `l` be the length of the list determined by calling `length` on `v`.
-
-    3.  Let `h` be the number of non-rest elements preceding the rest element if
+    2.  Let `h` be the number of non-rest elements preceding the rest element if
         there is one, or the number of elements if there is no rest element.
 
-    4.  Let `t` be the number of non-rest elements following the rest element if
-        there is one, or zero otherwise.
+    3.  Let `t` be the number of non-rest elements following the rest element if
+        there is one, or `0` otherwise.
+
+    4.  Check the length. If `p` is empty or has any non-rest elements:
+
+        1.  Let `l` be the length of the list determined by calling `length` on
+            `v`. *We only call `length` on the list if needed.*
+
+        2.  If `p` has both a rest element and `h + t > 0`:
```

Comment:
Nit: to me, the word "both" here is confusing.  It seems to imply that I'm supposed to read what follows as two things that `p` should have.  As in, "If `p` has a rest element and `p` has `h + t > 0`", which doesn't make any sense.

I think it reads better as just "If `p` has a rest element and `h + t > 0`"